### PR TITLE
Optimization: clientStreamInit() copied ClientStreamData

### PR DESCRIPTION
**Pull request by kinkie**
Detected by Coverity. CID 1529543 and CID 1529594: Unnecessary object
copies can affect performance (COPY_INSTEAD_OF_MOVE).

### CODE REVIEW

**Review by rousskov — CHANGES_REQUESTED** [review]
Please apply applicable #1797 concerns to this PR. Please do not post new PRs adding std::move wrappers until the affected PRs land.

File: src/client_side.cc
Comment (yadij) [inline src/client_side.cc:1354]: seems to be an unnecessary change.

File: src/client_side.cc
Comment (yadij) [inline src/client_side.cc:1360]: Is this type change necessary? if not please undo

**Review by yadij — CHANGES_REQUESTED** [review]

File: src/client_side.cc
Comment (kinkie) [reply src/client_side.cc:1354]: As I'm touching code near here, I applied AAA. I can revert if it's a problem

File: src/clientStream.cc
Comment (rousskov) [inline src/clientStream.cc:113]: If possible, please avoid readdata copies as well (Coverity defect 1529543?).

**Review by rousskov — DISMISSED** [review]
Thank you for optimizing this code. I am approving this PR after adjusting PR title/description and addressing Amos concerns, but I think we should do one more (similar) change in this PR.

File: src/clientStream.cc
Comment (kinkie) [reply src/clientStream.cc]: done

Comment (rousskov) [reply src/clientStream.cc]: Thank you. To reflect your changes, I have added CID 1529543 to the PR description.

**Review by rousskov — APPROVED** [review]

**Comment by kinkie** [comment]
@yadij your concerns should have been addressed. Mind re-reviewing here? Thanks!

**Review by yadij — APPROVED** [review]

**Pull request closed without merging**
